01_fib/srini/main.go

```diff
@@ -0,0 +1,39 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var outWriter io.Writer = os.Stdout
+
+func calculateFib(n int) []int {
+	//need to check for edge case to avoid array out of range
+	if n < 0 {
+		n = 0
```

Comment:
or you just abort here and return `[]int{}` for `n <= 0`

Reply:
fixed this